openerp/addons/base/ir/ir_ui_view.py

```diff
@@ -529,11 +529,19 @@ def apply_inheritance_specs(self, cr, uid, source, specs_tree, inherit_id, conte
                         node.getparent().remove(node)
                 elif pos == 'attributes':
                     for child in spec.getiterator('attribute'):
-                        attribute = (child.get('name'), child.text or None)
-                        if attribute[1]:
-                            node.set(attribute[0], attribute[1])
-                        elif attribute[0] in node.attrib:
-                            del node.attrib[attribute[0]]
+                        attribute = child.get('name')
+                        value = child.text or None
+                        if child.get('add') or child.get('remove'):
+                            value = node.get(attribute, "")
+                            assert not child.text
+                        if child.get('add'):
+                            value = "%s %s" % (value, child.get('add').strip())
+                        if child.get('remove'):
+                            value = " ".join([val for val in value.split(" ") if (" %s " % val) not in (" %s " % child.get('remove'))])
```

Comment:
Why not just use set operations there? e.g. 
```python
to_add = set(child.get('add', '').split())
to_remove = set(child.get('remove', '').split()
value = ' '.join(set(value.split()) + add - remove)
```

Also note:
```python
>>> ''.split()
[]
>>> ''.split(' ')
['']
```
so to split on whitespace, *not* providing a separator can be better than providing one.